### PR TITLE
Project controller spec part 8

### PR DESCRIPTION
**Pull request by mdeniz**

### CODE REVIEW

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 1]
```diff
@@ -875,4 +875,15 @@
       it { expect(assigns(:project).repositories.first.repository_architectures.count).to eq(Architecture.available.count) }
     end
   end
+
+  describe 'GET #remove_target_request_dialog' do
+    it "will success with an ajax call" do
```
Comment (bgeuken) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: succeed

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: :scream:

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 2]
```diff
@@ -799,4 +799,129 @@
       it { expect(user.home_project.repositories.find_by(name: 'valid_name').repository_architectures.count).to eq(1) }
     end
   end
+
+  describe 'POST #save_distributions' do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Isn't that what @ChrisBr is moving around currently?

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: :scream:  yes, you are right.... f*ck

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: @hennevogel  Talked to @ChrisBr at IRC about it and we agreed on merging this... after he rebase and add his own tests he can get rid of this... sorry!

Comment (hennevogel) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: so @mdeniz  this PR can be closed right?

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: @hennevogel Not closed!... should be merged. It have more things that the colliding things. And we also agreed in having this right now merged and then @ChrisBr can re-use it or get rid of.

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 3]
```diff
@@ -799,4 +799,129 @@
...
+    context "with a distribution but without target repository" do
+      it "Raises an ActiveRecord::RecordNotFound exception" do
+        expect do
+          post :save_distributions, project: user.home_project,
+                                    distributions: [distribution_without_target_repo.reponame]
+        end.to raise_error ActiveRecord::RecordNotFound
+      end
+    end
+
+    context "with two distributions using same repository" do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: This is only testing the validation right? Why is this here and not in a model test?

File: src/api/spec/controllers/webui/project_controller_spec.rb [hunk 4]
```diff
@@ -799,4 +799,129 @@
...
+
+      it { expect(flash[:success]).to eq("Successfully added repositories") }
+      it { is_expected.to redirect_to(action: :repositories, project: user.home_project) }
+      it { expect(assigns(:project).repositories.count).to eq(1) }
+      it { expect(assigns(:project).repositories.first.name).to eq('images') }
+      it { expect(assigns(:project).repositories.first.repository_architectures.count).to eq(Architecture.available.count) }
+    end
+  end
+
+  describe 'GET #remove_target_request_dialog' do
```
Comment (hennevogel) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: What are you testing? Rails `respond_to`?

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Just covering the method... should I remove this? Is a trivial thing I know, but at least this is covered by some test :disappointed:

File: src/api/spec/controllers/webui/project_controller_spec.rb (hunk 3 above)
Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Oh! You are right... and this is already tested at Repository spec (line 7). I will delete this, thanks

File: src/api/spec/controllers/webui/project_controller_spec.rb (hunk 4 above)
Comment (hennevogel) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Covering trivial things in the test suite is what makes it clumsy IMHO

Comment (mdeniz) [inline src/api/spec/controllers/webui/project_controller_spec.rb]: Ok, I will remove this then :+1:

**Comment by hennevogel** [comment]
Okay :+1: now :-)
